src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1308,6 +1382,26 @@ FinishDistributedExecution(DistributedExecution *execution)
 }
 
 
+/*
+ * CloseConnections closes all the connections from the given sessionList.
+ */
+static void
+CloseConnections(List *sessionList)
+{
+	ListCell *sessionCell = NULL;
+
+	foreach(sessionCell, sessionList)
+	{
+		WorkerSession *session = lfirst(sessionCell);
+		MultiConnection *con = session->connection;
+
+		UnclaimConnection(con);
+		con->remoteTransaction.transactionState = REMOTE_TRANS_INVALID;
```

Comment:
do we need this line?

Reply:
The reason I have added this line is because when closing the connection there is this check:
https://github.com/citusdata/citus/blob/master/src/backend/distributed/transaction/remote_transaction.c#L763

Since we skip some parts in TransactionState for repartition queries, the state is not set to REMOTE_TRANS_INVALID, yet we try to delete something that does not exist so this was to bypass that if check. I realize that there could be a better solution, or if we keep this solution I should document that I guess.

Reply:
As far as I can tell, we only do CloseConnections when we are done executing tasks. I would expect the transaction state to be REMOTE_TRANS_INVALID unless something is wrong. Perhaps this should be an assert?